src/data/nat/mul_ind.lean

```diff
@@ -0,0 +1,120 @@
+/-
+Copyright (c) 2021 Eric Rodriguez. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Rodriguez
+-/
+import number_theory.padics.padic_norm
+
+/-!
+# Multiplicative induction principles for ℕ
+
+This file provides three (closely linked) induction principles for ℕ, commonly used for proofs
+about multiplicative functions, such as the totient function and the Möbius function.
+
+## Main definitions
+
+* `nat.rec_on_prime_pow`: Given `P 0, P 1` and a way to extend `P a` to `P (p ^ k * a)`, you can
+  define `P` for all natural numbers.
+* `nat.rec_on_coprime`: Given `P 0`, `P (p ^ k)` for all prime powers, and a way to extend `P a` and
+  `P b` to `P (a * b)` when `a, b` are coprime, you can define `P` for all natural numbers.
+* `nat.rec_on_coprime'`: Given `P 0`, `P 1`, and `P (p ^ k)` for positive prime powers, and a way to
+  extend `P a` and `P b` to `P (a * b)` when `a, b` are coprime,
+  you can define `P` for all natural numbers.
+* `nat.rec_on_completely_multiplicative`: Given `P 0`, `P 1`, `P p` for all primes, and a proof that
+  you can extend `P a` and `P b` to `P (a * b)`, you can define `P` for all natural numbers.
+
+## Implementation notes
+
+The proofs use `padic_val_nat`; however, we have `padic_val_nat p = nat.log p $ nat.gcd k (p ^ k)`
+for any `p ∣ k`, which requires far less imports - the API isn't there though; however, this is why
+it's in `data` even though we import `number_theory`; it's not a particularly deep theorem.
+
+-/
+
+namespace nat
+
+-- REVIEWERS: should this be here? or in `nat.prime`? or should I just replace this with `haveI`?
+instance fact.min_fac_prime (n : ℕ) : fact (prime (n + 2).min_fac) :=
+⟨min_fac_prime $ succ_succ_ne_one n⟩
+
+-- also not sure where these two should be
+@[simp] lemma dvd_div_iff {a b c : ℕ} (hbc : c ∣ b) : a ∣ b / c ↔ a * c ∣ b :=
+begin
+  obtain ⟨x, rfl⟩ := hbc,
+  obtain rfl | hc := eq_zero_or_pos c,
+  { simp },
+  { rw [mul_comm c x, mul_dvd_mul_iff_right hc.ne', nat.mul_div_cancel _ hc] },
+end
+
+@[simp] lemma one_lt_pow_iff {k n : ℕ} (h : 0 ≠ k) : 1 < n ^ k ↔ 1 < n :=
+begin
+  cases n,
+  { cases k; simp },
+  cases n,
+  { rw one_pow },
+  refine ⟨λ _, one_lt_succ_succ n, λ _, _⟩,
+  induction k with k hk,
+  { exact absurd rfl h },
+  induction k with k _,
+  { simp },
+  exact one_lt_mul (one_lt_succ_succ _).le (hk (succ_ne_zero k).symm),
+end
+
+-- REVIEWERS: I'm also not in love with the naming - any suggestions welcome.
```

Comment:
`rec_on_prime_pow` and `rec_on_coprime` I'm completely happy with.

Can't think of anything better than `rec_on_coprime'` (`rec_on_coprime_pos`? `rec_on_coprime_and_one`??). 

How about `rec_on_mul` for `rec_on_completely_multiplicative`?